### PR TITLE
Delete/Revoke a claim

### PR DESCRIPTION
**Pull request by hobbitronics**
## Add
- RevokeModal
- delete claim button

image
image

### CODE REVIEW

File: src/pages/policies/[policyId]/claims/[claimId].svelte [hunk 1]
```diff
@@ -152,6 +156,23 @@ const onSubmit = async () => {
   await submitClaim(claimId)
   setNotice(didNeedFile ? `Added replacement cost and ${oldUploadLabel}` : 'Submitted changes')
 }
+const onClosed = async (e: CustomEvent): Promise<void> => {
```
Comment (Schparky) [inline src/pages/policies/[policyId]/claims/[claimId].svelte:159]: Given the size of this page, should `onClosed` be more descriptive? It's less than obvious what is being closed.

File: src/pages/policies/[policyId]/claims/[claimId].svelte [hunk 2]
```diff
@@ -353,12 +380,20 @@ const isFileUploadedByPurpose = (purpose: ClaimFilePurpose, files: ClaimFile[]):
...
+      />
 
       {#if showUploadButton}
         <Button raised disabled={noFilesUploaded} on:click={onSubmit}>{uploadLabelForButton}</Button>
       {/if}
       <br />
     </Row>
   {/if}
+
+  <RevokeModal {claim} open={revokeModalOpen} on:closed={onClosed} />
```
Comment (Schparky) [inline src/pages/policies/[policyId]/claims/[claimId].svelte:398]: This should be disabled if `is_removable` is `false`

**Comment by Schparky** [comment]
> * evokeModal
> * delete claim button

I often have to do a double-take on these when you omit the "Add". I generally repeat the "Add" or "Remove" on each bullet.

File: src/components/ClaimActions.svelte [hunk 3]
```diff
@@ -122,7 +122,8 @@ const onDeny = () => dispatch('deny', message)
 {/if}
 {#if isMemberOfPolicy}
   {#if isEditable}
-    <Button on:click={on('edit')} outlined>Edit claim</Button>
+    <Button raised on:click={on('edit')}>edit claim</Button>
+    <Button outlined on:click={on('delete')}><span class="material-icons">delete</span></Button>
```
Comment (Schparky) [inline src/components/ClaimActions.svelte:126]: Should this button be disabled as well if `is_removable` is false?

**Review by Schparky — APPROVED** [review]

File: src/components/ClaimActions.svelte (hunk 3 above)
Comment (hobbitronics) [reply src/components/ClaimActions.svelte:126]: perhaps and maybe it it doesn't need to be inside the control block then.

**Pull request merged**